weather_test.go

```diff
@@ -0,0 +1,48 @@
+package weather_test
+
+import (
+	"os"
+	"testing"
+	"weather"
+)
+
+func TestConstructUrl(t *testing.T) {
+
+	token := "foo"
+	got, err := weather.Request([]string{"PATH", "rio", "de", "janeiro"}, token)
```

Comment:
This seems to conflate two things: getting the desired location out of `os.Args`, and constructing the weather request. We should probably do those in two steps:

```go
location, err := weather.ParseArgs([]string{...})
... // handle error
got, err := weather.Request(location, token)
```